src/couch_jobs/src/couch_jobs.erl

```diff
@@ -385,10 +385,13 @@ wait_any(Subs, State, Timeout0, ResendQ) when
     end.
 
 
-limit_timeout(Timeout) when is_integer(Timeout), Timeout < 16#FFFFFFFF ->
+limit_timeout(_Timeout, true) ->
+    infinity;
+
+limit_timeout(Timeout, false) when is_integer(Timeout), Timeout < 16#FFFFFFFF ->
     Timeout;
 
-limit_timeout(_Timeout) ->
+limit_timeout(_Timeout, false) ->
```

Comment:
From the description I think this is wrong? Are you missing a `Timeout > 0` in the previous clause's guards?

Reply:
In the previous clause we return `infinity` if `NoSched` option is `true`. But that value is not passed directly to `erlfdb:wait(...)` it is then compared with `UserTimeout` https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/couch_jobs/src/couch_jobs.erl#L336 and we pass that to `erlfdb:wait(...)`.

We basically say that when `NoSched` is `true` we shouldn't pay attention the `MaxSTime` parameter when deciding how long or if to timeout at all when we wait to accept jobs.

Reply:
From your description:

> When no_schedule option is used, time scheduling parameter is set to 0 always, and so in that case, we have to special-case the limit to return infinity.

I read that as that we should return `infinity` when `Timeout == 0`. This catchall case wouldn't do that because the previous clause catches zero (and all negative integers).

Reply:
My description was referencing the logic a few API calls higher and was describing why we need to this.

In this particular clause `limit_timeout` that wouldn't work since `limit_timeout` is called after we clipped the value to a minimum of 100 msec, so I would have to put in 100 msec or so there so I opted to make it an explicit parameter.

Reply:
Ah. I see that's from higher up. Never mind!